code/datums/elements/boilable.dm

```diff
@@ -0,0 +1,43 @@
+/**
+ * Makes a reagent boilable
+ */
+/datum/element/boilable
+	element_flags = ELEMENT_DETACH | ELEMENT_BESPOKE
+	id_arg_index = 2
+
+	/// The temperature the reagent boils at
+	var/boiling_temp = INFINITY
+	/// The path of the resulting gas
+	var/datum/gas/vapor_path
+
+/datum/element/boilable/Attach(datum/reagent/target, _temp, _gas)
+	if(!istype(target))
+		return ELEMENT_INCOMPATIBLE
+	if(!GLOB.meta_gas_info[_gas])
+		return ELEMENT_INCOMPATIBLE
+
+	boiling_temp = _temp
+	vapor_path = _gas
+	RegisterSignal(target, COMSIG_REAGENT_TEMP_CHANGE, .proc/try_boil)
+	. = ..()
+	try_boil(target, target.holder?.chem_temp)
+
+/datum/element/boilable/Detach(datum/source, force)
+	UnregisterSignal(source, COMSIG_REAGENT_TEMP_CHANGE)
+	return ..()
+
+/// Attempts to boil the source reagent
+/datum/element/boilable/proc/try_boil(datum/reagent/source, _temp)
+	SIGNAL_HANDLER
+	if(_temp < boiling_temp)
+		return NONE
+	var/datum/reagents/holder = source.holder
+	if(!(holder?.flags & (DRAINABLE)))
+		return NONE
+	var/turf/location = get_turf(holder.my_atom)
+	if(!location)
+		return NONE
+
+	location.atmos_spawn_air("[vapor_path]=[source.volume * REAGENT_MOLE_DENSITY];TEMP=[_temp]")
```

Comment:
It depends on how much smoke and/or foam you used to spread the plasma. If you used enough it's less, if you used too little it's more.

Reply:
Bassically it's a sort of middle ground for gas production since you currently can get varying yields depending on how much smoke you use to apply the reagents. I am removing the gas production on smoke to prevent infinite gas production with a smoke machine.